client.go

```diff
@@ -0,0 +1,127 @@
+package matrixfederation
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// A Client makes request to the federation listeners of matrix
+// homeservers
+type Client struct {
+	client http.Client
+}
+
+// UserInfo represents information about a user.
+type UserInfo struct {
+	Sub string `json:"sub"`
+}
+
+// NewClient makes a new Client
+func NewClient() *Client {
+	// TODO: Verify ceritificates
+	tripper := federationTripper{
+		transport: &http.Transport{
+			// Set our own DialTLS function to avoid the default net/http SNI.
+			// By default net/http and crypto/tls set the SNI to the target host.
+			// By avoiding the default implementation we can keep the ServerName
+			// as the empty string so that crypto/tls doesn't add SNI.
+			DialTLS: func(network, addr string) (net.Conn, error) {
+				rawconn, err := net.Dial(network, addr)
+				if err != nil {
+					return nil, err
+				}
+				// Wrap a raw connection ourselves since tls.Dial defaults the SNI
+				conn := tls.Client(rawconn, &tls.Config{
+					ServerName:         "",
+					InsecureSkipVerify: true,
```

Comment:
We should be checking that the TLS certificate we see here matches one of the allowed SHA-256 fingerprints for the server.

That might be a bit much for the initial implementation, but a ``TODO`` comment might be nice.